ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-materialized-table-table-joins.json

```diff
@@ -0,0 +1,466 @@
+        "CREATE TABLE OUTPUT WITH (WRAP_SINGLE_VALUE=false) AS SELECT t1.id, t1.name name FROM T1 JOIN T2 ON T1.id = T2.id;",
+        "SELECT * FROM OUTPUT;"
+      ],
+      "inputs": [
+        {"topic": "T1", "key": 0, "value": "a", "timestamp": 0},
+        {"topic": "T2", "key": 0, "value": "b", "timestamp": 10},
+        {"topic": "T1", "key": 0, "value": null, "timestamp": 20},
+        {"topic": "T2", "key": 0, "value": null, "timestamp": 30},
+        {"topic": "T1", "key": 0, "value": null, "timestamp": 40}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`T1_ID` BIGINT KEY, `NAME` STRING"}}
+        ]}
+      ]
+    },
+    {
+      "name": "table-table-table - inner-inner",
+      "statements": [
+        "CREATE TABLE T1 (ID INT PRIMARY KEY, V0 bigint) WITH (kafka_topic='left', value_format='JSON');",
+        "CREATE TABLE T2 (ID INT PRIMARY KEY, V0 bigint) WITH (kafka_topic='right', value_format='JSON');",
+        "CREATE TABLE T3 (ID INT PRIMARY KEY, V0 bigint) WITH (kafka_topic='right2', value_format='JSON');",
+        "CREATE TABLE OUTPUT as SELECT T1.ID, T1.V0, T2.V0, T3.V0 FROM T1 JOIN T2 ON T1.ID = T2.ID JOIN T3 ON T1.ID = T3.ID;",
+        "SELECT * FROM OUTPUT;"
+      ],
+      "inputs": [
+        {"topic": "left", "key": 0, "value": {"V0": 1}, "timestamp": 0},
+        {"topic": "right", "key": 0, "value": {"V0": 2}, "timestamp": 1},
+        {"topic": "right2", "key": 0, "value": {"V0": 3}, "timestamp": 2},
+        {"topic": "left", "key": 0, "value": {"V0": 4}, "timestamp": 1000}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 0, "value": {"T1_V0": 1, "T2_V0": 2, "T3_V0": 3}, "timestamp": 2},
+        {"topic": "OUTPUT", "key": 0, "value": {"T1_V0": 4, "T2_V0": 2, "T3_V0": 3}, "timestamp": 1000}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`T1_ID` INTEGER KEY, `T1_V0` BIGINT, `T2_V0` BIGINT, `T3_V0` BIGINT"}},
+          {"row":{"columns":[0, 4, 2, 3]}}
+        ]}
+      ]
+    },
+    {
+      "name": "table-table-table - inner-full",
+      "statements": [
+        "CREATE TABLE T1 (ID INT PRIMARY KEY, V0 bigint) WITH (kafka_topic='left', value_format='JSON');",
+        "CREATE TABLE T2 (ID INT PRIMARY KEY, V0 bigint) WITH (kafka_topic='right', value_format='JSON');",
+        "CREATE TABLE T3 (ID INT PRIMARY KEY, V0 bigint) WITH (kafka_topic='right2', value_format='JSON');",
+        "CREATE TABLE OUTPUT as SELECT ROWKEY AS ID, T1.V0, T2.V0, T3.V0 FROM T1 JOIN T2 ON T1.ID = T2.ID FULL JOIN T3 ON T1.ID = T3.ID;",
```

Comment:
What's the semantic difference between `FULL JOIN` and `OUTER JOIN`?

Reply:
technically, we only support `INNER`, `LEFT OUTER`, and `FULL OUTER` joins between tables. So, depends on what you mean by `OUTER JOIN`. Source: https://docs.ksqldb.io/en/latest/developer-guide/joins/join-streams-and-tables/#table-table-joins

Reply:
I was just confused to see `FULL JOIN` here (because elsewhere we use the full name `FULL OUTER JOIN`) and the name of the test is `inner-full` while other test names say `outer join`, and wonder if these are different. 

So is `OUTER JOIN` just a syntax sugar for `FULL OUTER JOIN`, similar to `JOIN` being a syntax sugar of `INNER JOIN`?

Reply:
yes, I believe that is correct